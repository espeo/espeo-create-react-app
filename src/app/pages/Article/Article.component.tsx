import React, { PureComponent } from 'react';
import dayjs from 'dayjs';

import {
  AppWrapper,
  TitleWrapper,
  Title,
  ContentWrapper,
  ArticleImage,
  ArticleBDate,
  ArticleBAuthor,
  ArticleBTitle,
  ArticleTitle,
  ArticleContent,
  ArticleBDescription,
} from '@core/components';

import { ArticleContainer, BackLink, BackLinkText, SourceLink } from './styles';

interface ArticleOwnProps {
  location: any;
  article: string;
}

type ArticleProps = ArticleOwnProps;

class Article extends PureComponent<ArticleProps> {
  render() {
    const { article } = this.props.location.state;

    return (
      <AppWrapper>
        <TitleWrapper>
          <Title text="Article Details" />
        </TitleWrapper>
        <ContentWrapper>
          <BackLink to="/">
            <BackLinkText>Back to Articles</BackLinkText>
          </BackLink>
          <ArticleContainer>
            <ArticleBTitle>{article.title}</ArticleBTitle>
            <ArticleBDescription>{article.description}</ArticleBDescription>
            <ArticleImage alt={article.title} src={article.urlToImage} />
            <ArticleBDate>
              {dayjs(article.publishedAt).format('YYYY, MMM DD ')}
            </ArticleBDate>
            <ArticleBAuthor>{article.author}</ArticleBAuthor>
            <ArticleTitle>{article.title}</ArticleTitle>
            <ArticleContent>{article.content}</ArticleContent>
            <SourceLink href={article.url} target="_blank">
              Read more here &gt; Source: {article.source.name}
            </SourceLink>
          </ArticleContainer>
        </ContentWrapper>
      </AppWrapper>
    );
  }
}

export default Article;
