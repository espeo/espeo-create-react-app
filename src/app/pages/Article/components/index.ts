export * from './ArticleContainer';
export * from './BackLink';
export * from './SourceLink';
