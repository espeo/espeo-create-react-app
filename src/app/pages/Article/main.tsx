import React, { PureComponent } from 'react';
import {
  AppWrapper,
  TitleWrapper,
  Title,
  ContentWrapper,
} from '@core/components';

interface ArticleOwnProps {
  location: any;
  article: string;
}

type ArticleProps = ArticleOwnProps;

class Article extends PureComponent<ArticleProps> {
  render() {
    const { article } = this.props.location.state;
    console.log(article);

    return (
      <AppWrapper>
        <TitleWrapper>
          <Title text="title" />
        </TitleWrapper>
        <ContentWrapper></ContentWrapper>
      </AppWrapper>
    );
  }
}

export default Article;
