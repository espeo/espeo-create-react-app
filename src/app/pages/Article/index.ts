import { connect } from 'react-redux';
import { compose } from 'redux';

import Article from './components/ArticleComponent/Article.component';

export default compose(connect())(Article);
