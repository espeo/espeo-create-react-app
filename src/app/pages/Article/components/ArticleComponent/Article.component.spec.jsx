import renderer from 'react-test-renderer';
import React from 'react';
import { compose } from 'redux';
import { shallow, mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import {
    withProviders,
    withAdditionalProperties,
} from '@core/tests/theme-helpers';

import Article from './Article.component';

const locationMock = {
    state: {
        article: {
            author: 'author',
            content: 'content',
            description: 'description',
            publishedAt: 'publishedAt',
            source: {
                id: 'id',
                name: 'name',
            },
            id: 'id',
            name: 'name',
            title: 'title',
            url: 'url',
            urlToImage: 'urlToImage',
        }
    }
};

const WrappedComponent = compose(
    withProviders,
    withAdditionalProperties({
        location: locationMock
    })
)(Article);

describe('Article component test suite', () => {
    it('should match snapshot', () => {
        const component = renderer.create(<BrowserRouter><WrappedComponent /></BrowserRouter>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should mount article with children', () => {
        const text = 'Article Details';
        const wrapper = mount(<BrowserRouter><WrappedComponent /></BrowserRouter>);
        expect(wrapper.find('h1').text()).toEqual(text);
    });

    it('should render article component', () => {
        const wrapper = shallow(<WrappedComponent />);
        expect(wrapper.contains('.article'));
      });
});
