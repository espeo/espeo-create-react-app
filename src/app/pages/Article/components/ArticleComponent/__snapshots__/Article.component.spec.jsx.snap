// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component test suite should match snapshot 1`] = `
<div
  className="sc-htpNat cTiQRV article"
>
  <div
    className="sc-ifAKCX kjOYKb"
  >
    <h1
      className="sc-jzJRlG gUwiRh"
    >
      Article Details
    </h1>
  </div>
  <div
    className="sc-bxivhb dKaubX"
  >
    <a
      className="sc-kAzzGY knAwDu"
      href="/"
      onClick={[Function]}
    >
      <p
        className="sc-chPdSV dmrJyt"
      >
        Back to Articles
      </p>
    </a>
    <div
      className="sc-cSHVUG heEdLu article-content"
    >
      <p
        className="sc-VigVT sc-jTzLTM lkNGIF"
      >
        title
      </p>
      <p
        className="sc-bZQynM sc-gzVnrw fUMedX"
      >
        description
      </p>
      <img
        alt="title"
        className="sc-EHOje iXbLk"
        src="urlToImage"
      />
      <p
        className="sc-dnqmqq sc-iwsKbI iTtTSh"
      >
        Invalid Date
      </p>
      <p
        className="sc-dnqmqq sc-gqjmRU jBFvQt"
      >
        author
      </p>
      <p
        className="sc-VigVT dqedxV"
      >
        title
      </p>
      <p
        className="sc-bZQynM sc-htoDjs gWbtok"
      >
        content
      </p>
      <a
        className="sc-kGXeez dyIyAp"
        href="url"
        rel="noopener"
        target="_blank"
      >
        Read more here &gt; Source: name
      </a>
    </div>
  </div>
</div>
`;
